sa/storage-authority.go

```diff
@@ -253,20 +235,25 @@ func (ssa *SQLStorageAuthority) GetValidAuthorizations(ctx context.Context, regi
 		AND expires > ?
 		AND identifier IN (`+strings.Join(qmarks, ",")+`)
 		AND status = 'valid'
-		ORDER BY expires ASC
 		`, append([]interface{}{registrationID, now}, params...)...)
 	if err != nil {
 		return nil, err
 	}
 
 	byName := make(map[string]*core.Authorization)
 	for _, auth := range auths {
+		// No real life authorizations should have a nil expires. If we find them,
```

Comment:
Nit: On the DB server we know this is slow (because of an optimization problem), but it could still be large / slow here, too. The difference being we have no metrics on the loop size. Sadly, SQLStorageAuthority does not have a metrics object handy to record performance here. I think that's too big to do in this patch, and I leave this as a nit; you can open an issue or not as you see fit.